cpp/src/parquet/thrift_internal.h

```diff
@@ -256,6 +256,8 @@ static inline format::Type::type ToThrift(Type::type type) {
 static inline format::ConvertedType::type ToThrift(ConvertedType::type type) {
   // item 0 is NONE
   DCHECK_NE(type, ConvertedType::NONE);
+  // it is forbidden to emit "NA" (PARQUET-1990)
+  DCHECK_NE(type, ConvertedType::NA);
```

Comment:
DCHECK_Lt might be better here.

Reply:
I don't think it would change anything. Or we would have to do a full-blown check for valid values, which is a bit out of scope.

Reply:
OK.

Reply:
Ah, I can add that as well :-)